tests/api_fastapi/core_api/routes/ui/test_structure.py

```diff
@@ -404,7 +405,7 @@ class TestStructureDataEndpoint:
     def test_should_return_200(self, test_client, params, expected):
         response = test_client.get("/ui/structure/structure_data", params=params)
         assert response.status_code == 200
-        assert response.json() == expected
+        assert not DeepDiff(response.json(), expected, ignore_order=True)
```

Comment:
Can we just sort by key instead?

Reply:
Problem is its nested map of array, its not working with sort_keys.

Reply:
or else we need to build sort function and sort all the nested keys then assert, dont think its worth here? DeepDiff is simple to assert

Reply:
Yeah looks cool

Reply:
Why rebuild the wheel!

Reply:
BTW DeepDiff internally doing sorting to compare.

Reply:
Nice small tool.

BTW. @gopidesupavan -> maybe we could use `deephash` from the same set of tools to make stable hashes of config dictionaries? 

There is one thing that is "specific" to how config help is extracted (Python 3.12 performs slightly different dedenting of docstrings extracted from methods) - but I think `deephash` should nicely handle the last change with opts sorting and possibly other in-stabilities that `click` might introduce - see https://github.com/apache/airflow/pull/45156

Reply:
Yeah it helps they have some useful tools. https://zepworks.com/deepdiff/current/. 

Yes deepdiff really useful and nice tool. we use this tool more in our work to compare terraform plans(its a huge nested dict :) ). with old state and new states change to extract some data for triage.